File: 201-300/242/clickmouse.c
void PerCoLateDown(char* s, int i, int N){
    char X = s[i];
    int child;
    for(;2*i+1 <= N-1; i = child){
    	child = 2*i+1;
        if(child != N-1 && s[child+1] > s[child])
            child++;
        if(s[child] > X) 
            s[i] = s[child];
        else break;
    }
    s[i] = X;
}
void Sort( char* s ){
    int i;
    int N;
    for(i = 0; s[i] != '\0'; i++);
    N = i;
    for(i = N/2-1; i >= 0; i--)
        PerCoLateDown(s, i, N);
    for(i = N-1; i >= 0 ; i--){
        int temp = s[i];
        s[i] = s[0];
        s[0] = temp;
        PerCoLateDown(s, 0, i);
    }
}
bool isAnagram(char* s, char* t) {
    Sort(s);
    Sort(t);
    if(strcmp(s, t) == 0)
        return true;
    else
        return false;
}

File: 201-300/242/clickmouse.java
class Solution {
    public boolean isAnagram(String s, String t) {
        int[] A = new int[26];
        int i;
        for(i = 0; i < s.length(); i++)
            A[s.charAt(i)-'a']++;
        for(i = 0; i < t.length(); i++)
            A[t.charAt(i)-'a']--;
        for(i = 0; i < 26; i++)
            if(A[i] != 0)
                break;
        if(i < 26)
            return false;
        else
            return true;
    }
}
